Add recipe for org-auto-export-pandoc

Y0ngg4n opened the pull request:
### Brief summary of what the package does
This package adds the posibility to add the metadata tag `#+auto-export-pandoc: to-markdown` to org-mode files to automatically export the org-mode file via pandoc in the corresponding format. It just calls the corresponding function.
I am new to elisp, i hope i did nothing wrong. If this is not worth to be a package then please close it, but i experienced it as very useful.

### Direct link to the package repository

https://github.com/Y0ngg4n/org-auto-export-pandoc

### Your association with the package

Maintainer

### Relevant communications with the upstream package maintainer
None needed

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- []x My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this.  I think you are being too modest when saying this might not be useful to others, but if you're ever uncertain then another (better?) place may be a public forum like #emacs on libera.chat, or the Emacs subreddit, etc.  Here's a quick first pass -

⸺ `org-auto-export-pandoc.el` with byte-compile using Emacs 29.4:
```
org-auto-export-pandoc.el:16:1: Warning: reference to free variable `…'
org-auto-export-pandoc.el:30:2: Error: Symbol's value as variable is void: ox-pandoc
```

⸺ `org-auto-export-pandoc.el` with package-lint 20240923.1044:
```
4 issues found:
1:0: warning: "Version:" or "Package-Version:" header is missing. MELPA will handle this, but other archives will not.
1:0: error: package.el cannot parse this buffer: Search failed: ";;; org-auto-pandoc.el ends here"
1:73: warning: You should depend on (emacs "24.1") if you need lexical-binding.
19:22: warning: Use a properly versioned dependency on "ox-pandoc" if possible.
```

⸺ `org-auto-export-pandoc.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- The `;;; Commentary` for this file is much wider than 80 characters
- org-auto-export-pandoc.el#L46: It's safer to sharp-quote function names; use `#'`
- org-auto-export-pandoc.el#L46: Loading a package should rarely add hooks
```

`#'load`-check on each file:
```
Loading org-auto-export-pandoc.el
  org-auto-export-pandoc.el:Error: Emacs 29.4:
  (void-variable …)
```

⸺ Package and license:
- Please specify `:fetcher` before `:repo` in your recipe
- Unreachable package URL in org-auto-export-pandoc.el: https://example.com/foo
- org-auto-export-pandoc.el needs *formal* license boilerplate and/or an [SPDX-License-Identifier](https://spdx.dev/ids/)

Y0ngg4n commented:
@riscy thank you for the review. I will fix it.

Y0ngg4n commented:
@riscy I fixed the stuff. The only thing i don´t want is to stick to a specific ox-pandoc version as it is just a wrapper around ox-pandoc.

riscy commented:
Thanks!  Here's the latest -

⸺ `org-auto-export-pandoc.el` with package-lint 20240923.1044:
```
1 issue found:
20:22: warning: Use a properly versioned dependency on "ox-pandoc" if possible.
```
I don't understand the rationale re: "it is just a wrapper around ox-pandoc".  Even as a wrapper, it must depend on some specific version of ox-pandoc, because ox-pandoc can change over time.  Here's a link to ox-pandoc on the MELPA stable: https://stable.melpa.org/#/ox-pandoc

These are the same as before:
- The `;;; Commentary` for this file is much wider than 80 characters - and while I'm on this, the abbreviated(?) license boilerplate you have ";; This file is free software… along with this file.  If not, see <https://www.gnu.org/licenses/>." seems to be serving no purpose whatsoever - maybe you can clarify?
- org-auto-export-pandoc.el#L48: It's safer to sharp-quote function names; use `#'`
- org-auto-export-pandoc.el#L48: Loading a package should rarely add hooks - let the user add this hook if they want, as one way to use your package.  The might alternatively wand to bind this to a key, or add a hook on some other event.
```

Y0ngg4n commented:
@riscy thank you for your patience. I am sorry i am new to elisp and not very good at it. 

- I removed the License Boilerplate as the licence is already in the repository and specified by `SPDX-License-Identifier`.
- Hm sorry i don´t get it how to make this.
Do you mean: `(funcall #'(intern (concat "org-pandoc-export-" export-format)))`?
I always get `Unused lexical variable export-format` with this.
- I removed the hook and added it to the commentary and the readme

riscy commented:
> Hm sorry i don´t get it how to make this.

I think this may have actually been a false positive on my side, since you're doing some metacody `(funcall (intern (concat ...` stuff that my admittedly primitive linter struggles with.  Nevertheless it's not an issue with the current code.

I was just scanning the MELPA index for anything similar and it looks like I overlooked https://melpa.org/#/org-autoexport - have you seen this package?  Is it compatible with pandoc?

Y0ngg4n commented:
@riscy as far as i can read in the code it only supports ghf markdown and latex but not pandoc

riscy commented:
Sorry to ping pong this back and forth so many times, but what I should have (also) asked was, is there any possibility of you integrating some of your improvements (and your support for pandoc) into that existing package?  If that's a *possibility*, this might be better for users overall, since they'd have a unified interface to autoexporting.

If not, though, the code looks good and is ready for a quick merge.

Y0ngg4n commented:
@riscy yeah that would be probably better. I will try to dig into the code of it.

Y0ngg4n commented:
@riscy I have contacted the maintainer.
Here is the answer: https://todo.sr.ht/~zondo/org-autoexport/3#event-395431
So i guess we can merge this.

riscy commented:
Hi, I see -

> the pandoc exporter is a special case that breaks my assumption ... I think it's not much of a problem if a separate package exists to handle the pandoc special case ... I can document in my package that pandoc is special, and point people at your package instead.
>
> That said, if you have a patch which you think will integrate with org-autoexport, then I'll be glad to check it out.

I think it would be great if both packages mention each other in their READMEs (especially describing pandoc as a special case), and strive toward similar interfaces or even merging together.  But I'm fine with merging this in that case - and thank you for pursuing that lead like you did!

One last reminder: as we hit merge, ensure GitHub release 0.0.1 is up-to-date with your current code and `Package-Version`

That said, welcome to MELPA. :)

zondo commented:
FYI, I've added a warning to the org-autoexport README: https://git.sr.ht/~zondo/org-autoexport#usage which points at the org-auto-export-pandoc package for the pandoc case.

Y0ngg4n commented:
@riscy @zondo thank you both!
